pkg/types/stream.go

```diff
@@ -26,15 +26,31 @@ var defaultDialer = &websocket.Dialer{
 
 type Stream interface {
 	StandardStreamEventHub
+	BasicSubscription
 
-	Subscribe(channel Channel, symbol string, options SubscribeOptions)
-	GetSubscriptions() []Subscription
+	// SetPublicOnly connects to public or private
 	SetPublicOnly()
 	GetPublicOnly() bool
+
+	// Connect connects to websocket server
 	Connect(ctx context.Context) error
+	Reconnect()
 	Close() error
 }
 
+type BasicSubscription interface {
+	// Subscribe subscribes the specific channel.
+	Subscribe(channel Channel, symbol string, options SubscribeOptions)
+	GetSubscriptions() []Subscription
+	// ClearSubscriptions clears the subscriptions
+	ClearSubscriptions()
```

Comment:
In fact, ClearSubscriptions does not change any thing from the server side, it's related to Resubscribe(), we could merge this call to other Resubscribe() things